src/ServiceInsight/Search/SearchBarViewModel.cs

```diff
@@ -91,22 +90,27 @@ public void Search()
         {
             SearchInProgress = true;
             AddRecentSearchEntry(SearchQuery);
-            Parent.RefreshMessages(SelectedEndpoint, 1, SearchQuery);
+            Parent.Search(SelectedEndpoint, SearchQuery);
         }
 
         public void CancelSearch()
         {
             SearchQuery = null;
             SearchInProgress = false;
-            Parent.RefreshMessages(SelectedEndpoint, 1, SearchQuery);
+            Parent.Search(SelectedEndpoint, SearchQuery);
         }
 
-        public void SetupPaging(PagedResult<StoredMessage> pagedResult)
+        public void SetupPaging(PagedResult<StoredMessage> pagedResult, string selfLink)
         {
-            Result = pagedResult.Result;
             CurrentPage = pagedResult.TotalCount > 0 ? pagedResult.CurrentPage : 0;
             TotalItemCount = pagedResult.TotalCount;
-
+            Result = pagedResult.Result;
+            NextLink = pagedResult.NextLink;
+            PrevLink = pagedResult.PrevLink;
+            FirstLink = pagedResult.FirstLink;
+            LastLink = pagedResult.LastLink;
+            PageSize = pagedResult.PageSize;
+            SelfLink = selfLink;
```

Comment:
This is the only reference to the `SelfLink` property, i.e. it is never read, only written to. Can it be removed?

If so, the `selfLink` parameter can also be removed from this method.